packages/framework-provider-local-infrastructure/src/test-helper/local-test-helper.ts

```diff
@@ -0,0 +1,43 @@
+import { LocalQueries } from './local-queries'
+import { LocalCounters } from './local-counters'
+
+interface ApplicationOutputs {
+    graphqlURL: string
+    websocketURL: string
+}
+
+export class LocalTestHelper {
+    private constructor(
+        readonly outputs: ApplicationOutputs,
+        readonly counters: LocalCounters,
+        readonly queries: LocalQueries
+    ) {}
+
+    public static async build(appName: string): Promise<LocalTestHelper> {
+        this.ensureProviderIsReady()
+        return new LocalTestHelper(
+          {
+            graphqlURL: await this.graphqlURL(),
+            websocketURL: await this.websocketURL(),
+          },
+          new LocalCounters(`${appName}-app`),
+          new LocalQueries()
+        )
+      }
+    
+      private static ensureProviderIsReady(): void {
+        if (false) {
+          throw new Error(`The local provider is not ready.`)
+        }
+      }         
```

Comment:
If there is nothing to do to ensure the provider is ready, I would remove the function. 

I will definitely avoid leaving this kind of code in `main`, as it is highly confusing. 
We should try to leave the `main` branch as clean as possible.